torchgeo/datasets/substation.py

```diff
@@ -0,0 +1,247 @@
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, download_url, extract_archive
+
+
+class Substation(NonGeoDataset):
+    """Class for Substation Dataset.
+
+    The `Substation <https://github.com/Lindsay-Lab/substation-seg/blob/main/README.md>`__
+    dataset is responsible for handling the loading and transformation of
+    substation segmentation datasets. It extends NonGeoDataset, providing methods
+    for dataset verification, downloading, and transformation.
+    Dataset Format:
+    * .npz file for each datapoint
+
+    Dataset Features:
+
+    * 26,522 image-mask pairs stored as numpy files.
+    * Data from 5 revisits for most locations.
+    * Multi-temporal, multi-spectral images (13 channels) paired with masks,
+      with a spatial resolution of 228x228 pixels
+
+    If you use this dataset in your research, please cite the following:
+    * https://doi.org/10.48550/arXiv.2409.17363
+    """
+
+    directory = 'Substation'
+    filename_images = 'image_stack.tar.gz'
+    filename_masks = 'mask.tar.gz'
+    url_for_images = 'https://storage.googleapis.com/tz-ml-public/substation-over-10km2-csv-main-444e360fd2b6444b9018d509d0e4f36e/image_stack.tar.gz'
+    url_for_masks = 'https://storage.googleapis.com/tz-ml-public/substation-over-10km2-csv-main-444e360fd2b6444b9018d509d0e4f36e/mask.tar.gz'
+
+    def __init__(
+        self,
+        root: Path,
+        bands: list[int],
+        mask_2d: bool,
+        timepoint_aggregation: str = 'concat',
+        download: bool = False,
+        checksum: bool = False,
+        num_of_timepoints: int = 4,
+        use_timepoints: bool = False,
+    ) -> None:
+        """Initialize the Substation.
+
+        Args:
+            root: Path to the directory containing the dataset.
+            bands: Number of channels to use from the image.
+            use_timepoints: Whether to use multiple timepoints for each image.
+            mask_2d: Whether to use a 2D mask.
+            timepoint_aggregation: How to aggregate multiple timepoints.
+            num_of_timepoints: Number of timepoints to use for each image.
+            download: Whether to download the dataset if it is not found.
+            checksum: Whether to verify the dataset after downloading.
+        """
+        self.root = root
+        self.bands = bands
+        self.use_timepoints = use_timepoints
+        self.timepoint_aggregation = timepoint_aggregation
+        self.mask_2d = mask_2d
+        self.image_dir = os.path.join(root, 'image_stack')
+        self.mask_dir = os.path.join(root, 'mask')
+        self.num_of_timepoints = num_of_timepoints
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+        self.load_image_filenames()
+
+    def load_image_filenames(self) -> None:
+        """Load image filenames from the image directory."""
+        self.image_filenames = os.listdir(self.image_dir)
```

Comment:
```suggestion
        self.image_filenames = sorted(os.listdir(self.image_dir))
```
No need for a helper function for a single line of code. Also, sorting this list makes the dataset reproducible.